<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>607</width>
    <height>338</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="widget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="ModelRendering" name="viewer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>292</width>
        <height>0</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5"/>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Focal Length (cm)</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="focalLengthLineEdit">
          <property name="text">
           <string>40</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Mesh Height (cm)</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="meshHeightLineEdit">
          <property name="text">
           <string>10</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Mesh Width (cm)</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="meshWidthLineEdit">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>10</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>607</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoadModel"/>
    <addaction name="actionSaveModel"/>
    <addaction name="separator"/>
    <addaction name="actionLoadLightRayReceiverPosition"/>
    <addaction name="separator"/>
    <addaction name="actionSave_Vertices"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionGenerateTriangles"/>
   </widget>
   <widget class="QMenu" name="menuAlgorithm">
    <property name="title">
     <string>Algorithm</string>
    </property>
    <addaction name="actionRunTargetOptimization"/>
    <addaction name="actionShoot_Test_Ray"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionNormals"/>
    <addaction name="actionAxis"/>
    <addaction name="actionDesired_Normals"/>
    <addaction name="actionDesired_Ray_Directions"/>
    <addaction name="actionDesired_Rays"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuAlgorithm"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionSaveModel">
   <property name="text">
    <string>Save Model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionLoadModel">
   <property name="text">
    <string>Load Model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionGenerateTriangles">
   <property name="text">
    <string>Generate Triangles</string>
   </property>
  </action>
  <action name="actionSave_Vertices">
   <property name="text">
    <string>Save Vertices</string>
   </property>
  </action>
  <action name="actionRunTargetOptimization">
   <property name="text">
    <string>Run Target Optimization</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionLoadLightRayReceiverPosition">
   <property name="text">
    <string>Load Light-Ray Receiver Position</string>
   </property>
  </action>
  <action name="actionNormals">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Normals</string>
   </property>
  </action>
  <action name="actionAxis">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Axis</string>
   </property>
  </action>
  <action name="actionDesired_Normals">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Desired Normals</string>
   </property>
   <property name="toolTip">
    <string>DesiredNormals</string>
   </property>
  </action>
  <action name="actionDesired_Ray_Directions">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Desired Ray Directions</string>
   </property>
  </action>
  <action name="actionDesired_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Desired Rays</string>
   </property>
  </action>
  <action name="actionShoot_Test_Ray">
   <property name="text">
    <string>Shoot Test-Ray</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ModelRendering</class>
   <extends>QWidget</extends>
   <header>rendering.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
